### PR TITLE
Azure App Service Manage for Slot

### PR DESCRIPTION
**Pull request by vincentdass**

### CODE REVIEW

File: Tasks/AzureAppServiceManage/task.json [hunk 1]
```diff
@@ -56,6 +55,13 @@
             "helpMarkDown": "Enter or select the name of an existing Azure App Service"
         },
         {
+            "name": "SpecifySlot",
+            "type": "boolean",
+            "label": "Specify Slot",
```
Comment (mvvsubbu) [inline Tasks/AzureAppServiceManage/task.json]: Please get this label reviewed by PM

File: Tasks/Common/azurerest-common/azurerestutility.ts [hunk 2]
```diff
@@ -313,86 +313,89 @@ export async function swapWebAppSlot(SPN, resourceGroupName: string, webAppName:
...
             tl.error(response.statusMessage);
-            deferred.reject(tl.loc("FailedtoStartAppService",webAppName, response.statusCode, response.statusMessage));
+            deferred.reject(tl.loc("FailedtoStartAppService",webAppNameWithSlot, response.statusCode, response.statusMessage));
         }
     });
     return deferred.promise;
 }
 
-export async function stopAppService(SPN, resourceGroupName: string, webAppName: string) {
+export async function stopAppService(SPN, resourceGroupName: string, webAppName: string, specifySlotFlag: boolean, slotName: string) {
```
Comment (mvvsubbu) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: Should we call input as endPoint rather than SPN

File: Tasks/AzureAppServiceManage/task.loc.json [hunk 3]
```diff
@@ -54,6 +54,13 @@
       "helpMarkDown": "ms-resource:loc.input.help.WebAppName"
     },
     {
+      "name": "SpecifySlot",
```
Comment (mvvsubbu) [inline Tasks/AzureAppServiceManage/task.loc.json]: change "name": "OperateOnSlot"

File: Tasks/AzureAppServiceManage/task.json (hunk 1 above)
Comment (vincentdass) [inline Tasks/AzureAppServiceManage/task.json]: task.json is created as per spec.

File: Tasks/Common/azurerest-common/azurerestutility.ts [hunk 4]
```diff
@@ -313,86 +313,89 @@ export async function swapWebAppSlot(SPN, resourceGroupName: string, webAppName:
...
             deferred.reject(error);
         }
         if(response.statusCode === 200 || response.statusCode === 204) {
-            deferred.resolve(tl.loc('AppServicestartedsuccessfully', webAppName));
+            deferred.resolve(tl.loc('AppServicestartedsuccessfully', webAppNameWithSlot));
         }
         else {
             tl.error(response.statusMessage);
-            deferred.reject(tl.loc("FailedtoStartAppService",webAppName, response.statusCode, response.statusMessage));
+            deferred.reject(tl.loc("FailedtoStartAppService",webAppNameWithSlot, response.statusCode, response.statusMessage));
```
Comment (vincentdass) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: done

File: Tasks/Common/azurerest-common/azurerestutility.ts [hunk 5]
```diff
@@ -313,86 +313,89 @@ export async function swapWebAppSlot(SPN, resourceGroupName: string, webAppName:
     return deferred.promise;
 }
 
-export async function startAppService(SPN, resourceGroupName: string, webAppName: string) {
+export async function startAppService(SPN, resourceGroupName: string, webAppName: string, specifySlotFlag: boolean, slotName: string) {
```
Comment (vincentdass) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: done

File: Tasks/AzureAppServiceManage/task.loc.json (hunk 3 above)
Comment (mvvsubbu) [inline Tasks/AzureAppServiceManage/task.loc.json]: Keep back variable name in sync with value for name. If it is SpecifySlot keep the same thing

File: Tasks/AzureAppServiceManage/task.json [hunk 6]
```diff
@@ -18,8 +18,7 @@
   },
   "minimumAgentVersion": "1.102.0",
   "instanceNameFormat": "$(Action): $(WebAppName)",
-  "groups": [
-
+    "groups": [
```
Comment (Ajay-MS) [inline Tasks/AzureAppServiceManage/task.json]: Indentation

File: Tasks/AzureAppServiceManage/azureappservicemanage.ts [hunk 7]
```diff
@@ -1,97 +0,0 @@
-import tl = require('vsts-task-lib/task');
-import Q = require('q');
-import path = require('path');
-var azureRmUtil = require('azurerest-common/azurerestutility.js');
-var kuduLogUtil = require('azurerest-common/kududeploymentstatusutility.js');
```
Comment (mvvsubbu) [inline Tasks/AzureAppServiceManage/azureappservicemanage.ts]: why is this deleted?
